lib/services/ios-debug-service.ts

```diff
@@ -123,86 +123,99 @@ class IOSDebugService implements IDebugService {
 			}
 		});
 
-		await this.wireDebuggerClient(projectData);
+		return this.wireDebuggerClient(debugData, debugOptions);
 	}
 
-	private async emulatorStart(projectData: IProjectData): Promise<void> {
-		await this.wireDebuggerClient(projectData);
+	private async emulatorStart(debugData: IDebugData, debugOptions: IDebugOptions): Promise<string> {
+		const result = await this.wireDebuggerClient(debugData, debugOptions);
 
-		let attachRequestMessage = this.$iOSNotification.getAttachRequest(projectData.projectId);
+		let attachRequestMessage = this.$iOSNotification.getAttachRequest(debugData.applicationIdentifier);
 
 		let iOSEmulator = <Mobile.IiOSSimulatorService>this.$iOSEmulatorServices;
 		await iOSEmulator.postDarwinNotification(attachRequestMessage);
+		return result;
 	}
 
-	private async deviceDebugBrk(projectData: IProjectData, buildConfig: IBuildConfig, shouldBreak?: boolean): Promise<void> {
-		await this.$devicesService.initialize({ platform: this.platform, deviceId: this.$options.device });
-		this.$devicesService.execute(async (device: iOSDevice.IOSDevice) => {
+	private async deviceDebugBrk(debugData: IDebugData, shouldBreak: boolean, debugOptions: IDebugOptions): Promise<string[]> {
```

Comment:
isn't `shouldBreak` a debug option? Can't we place it in debugOptions object?
P.S. I do not like passing boolean arguments.